### PR TITLE
Refine type returns of createElement()

### PR DESCRIPTION
**Pull request by psam44**
For https://github.com/google/closure-library/issues/570

### CODE REVIEW

File: closure/goog/base.js [hunk 1]
```diff
@@ -1129,7 +1129,8 @@ if (goog.DEPENDENCIES_ENABLED) {
   goog.appendScriptSrcNode_ = function(src) {
     /** @type {Document} */
     var doc = goog.global.document;
-    var scriptEl = doc.createElement('script');
+    var scriptEl = /** @type {HTMLScriptElement} */
```
Comment (shicks) [inline closure/goog/base.js]: Would it make sense to mark these as {!HTMLScriptElement} ?

**Comment by shicks** [comment]
I'm not sure why the CLA check still fails for you, since you did respond in the last PR.  Maybe it's having trouble linking the address you signed with to your github username?

File: closure/goog/base.js (hunk 1 above)
Comment (psam44) [inline closure/goog/base.js]: You're probably right, because of {!Element} of Document.prototype.createElement.

**Comment by psam44** [comment]
CLA: yes, my primary google email and my github email are different.
Added the github account name, to see if it helps.

File: closure/goog/base.js (hunk 1 above)
Comment (shicks) [inline closure/goog/base.js]: Okay, please make that change and then we can pull this.

**Comment by concavelenz** [comment]
I'll pull this in manually.  This revealed an issue with the conformance framework, so it needed some work to land.  Should go out today.

**Comment by joeltine** [comment]
Merged in https://github.com/google/closure-library/commit/54b3bebb1698a0ad00ce3dce6a9850e381c7d521#diff-c359099ba4bc09bc2864536c49356d52

**Pull request closed without merging**